core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java

```diff
@@ -149,13 +149,14 @@ public synchronized void stop(long timeout, TimeUnit timeUnit) {
         try {
```

Comment:
Btw, looking at this, I noticed above code calling `stop()` never throws any exception, so the above catch blocks are unnecessary.